doc/reference/configuration/configuration_reference.rst

```diff
@@ -392,6 +392,293 @@ audit_log.syslog.*
+
+.. _configuration_reference_compat_replication_timeout:
+
+.. confval:: compat.box_cfg_replication_sync_timeout
+
+    Sets a default replication sync timeout: 0 (`new`) or 300 seconds (`old`)
+
+    .. important::
+
+        This value is set during the initial ``box.cfg{}`` call and cannot be changed later.
+
+    See also: :ref:`compat-option-replication-timeout`
+
+    |
+    | Type: string
+    | Possible values: 'new', 'old'
+    | Default: 'new'
+    | Environment variable: TT_COMPAT_BOX_CFG_REPLICATION_SYNC_TIMEOUT
+
+.. _configuration_reference_compat_error_serialize:
+
+.. confval:: compat.box_error_serialize_verbose
+
+    Controls the verbosity of :ref:`error objects <box_error-error_object>` serialization:
+
+    -   ``new``: serialize the error message together with other potentially useful fields
+    -   ``old``: serialize only the error message
+
+    |
+    | Type: string
+    | Possible values: 'new', 'old'
+    | Default: 'old'
+    | Environment variable: TT_COMPAT_BOX_ERROR_SERIALIZE_VERBOSE
+
+.. _configuration_reference_compat_error_unpack:
+
+.. confval:: compat.box_error_unpack_type_and_code
+
+    Whether to show error fields in ``box.error.unpack()``:
+
+    -   ``new``: do not show ``base_type`` and ``custom_type`` fields; do not show
+        the ``code`` field if it is 0. Note that ``base_type`` is still accessible for error object.
+    -   ``old``: show all fields
+
+    |
+    | Type: string
+    | Possible values: 'new', 'old'
+    | Default: 'old'
+    | Environment variable: TT_COMPAT_BOX_ERROR_UNPACK_TYPE_AND_CODE
+
+.. _configuration_reference_compat_cluster_meaning:
+
+.. confval:: compat.box_info_cluster_meaning
+
+    Whether ``box.info.cluster`` should show the current replica set (`old`) or
+    the whole cluster with all its replica sets (`new`).
+
+    |
+    | Type: string
+    | Possible values: 'new', 'old'
+    | Default: 'new'
+    | Environment variable: TT_COMPAT_BOX_INFO_CLUSTER_MEANING
+
+
+.. _configuration_reference_compat_session_push:
+
+.. confval:: compat.box_session_push_deprecation
+
+    Whether to raise errors on attempts to call the deprecated function ``box.session.push``:
+
+    -   ``new``: raise an error
+    -   ``old``: do not raise an error
+
+
+    |
+    | Type: string
+    | Possible values: 'new', 'old'
+    | Default: 'old'
+    | Environment variable: TT_COMPAT_BOX_SESSION_PUSH_DEPRECATION
+
+.. _configuration_reference_compat_space_execute:
+
+.. confval:: compat.box_space_execute_priv
+
+    Whether the ``execute`` privilege can be granted on spaces:
+
+    -   ``new``: an error is raised
+    -   ``old``: the privilege can be granted with no actual effect
+
+    |
+    | Type: string
+    | Possible values: 'new', 'old'
+    | Default: 'new'
+    | Environment variable: TT_COMPAT_BOX_SPACE_EXECUTE_PRIV
+
+.. _configuration_reference_compat_space_max:
+
+.. confval:: compat.box_space_max
+
+    Controls the max space id (``box.schema.SPACE_MAX``):
```

Comment:
```suggestion
    Set the maximum space identifier (``box.schema.SPACE_MAX``):
```

Let's also update a corresponding value in the `Limitations` topic: https://www.tarantool.io/en/doc/latest/book/box/limitations/